### PR TITLE
Adding Reseller component and REST APIs

### PR DESCRIPTION
**Pull request by rftemple**
Added Reseller component and REST APIS

Signed-off-by: Templeton, Randall <randall.f.templeton@intel.com>

### CODE REVIEW

**Review by trbehera — APPROVED** [review]

File: service/http-api-samples/http-api-keystore-sample/pom.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            <goals>
+             <goal>exec</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <executable>java</executable>
+          <includeProjectDependencies>true</includeProjectDependencies>
+          <workingDirectory>${project.build.outputDirectory}</workingDirectory>
+          <mainClass>org.fido.iot.sample.DiClientApp</mainClass>
```
Comment (trbehera) [inline service/http-api-samples/http-api-keystore-sample/pom.xml:104]: The exec main class is set as DiClientApp for every API class. This needs to be updated for API samples.

**Comment by Darshini-Parikh93** [comment]
Will there be a separate PR to enable docker deployment for reseller?

File: service/http-api-samples/http-api-rv-sample/pom.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    <dependency>
+      <groupId>org.fido.iot</groupId>
+      <artifactId>protocol</artifactId>
+      <version>0.3-SNAPSHOT</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.fido.iot</groupId>
+      <artifactId>storage-owner-sample</artifactId>
```
Comment (Darshini-Parikh93) [inline service/http-api-samples/http-api-rv-sample/pom.xml:44]: Shouldn't this be storage-rv-sample?

File: service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerConfigLoader.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      defaultConfiguration.addProperty(ResellerAppConstants.API_PORT, "8070");
+      defaultConfiguration.addProperty(ResellerAppConstants.DB_DRIVER, "org.h2.Driver");
+      defaultConfiguration.addProperty(ResellerAppConstants.SERVER_PATH,
+          System.getProperty(ResellerAppConstants.USER_DIR));
+      defaultConfiguration.addProperty(ResellerAppConstants.DB_USER, "sa");
+      defaultConfiguration.addProperty(ResellerAppConstants.DB_PWD, "");
+      defaultConfiguration.addProperty(ResellerAppConstants.DB_PORT, "8071");
+      defaultConfiguration.addProperty(ResellerAppConstants.KEYSTORE_TYPE, "PKCS11");
+      defaultConfiguration.addProperty(ResellerAppConstants.KEYSTORE_PATH, "");
+      defaultConfiguration.addProperty(ResellerAppConstants.KEYSTORE_PWD, "123456");
```
Comment (Darshini-Parikh93) [inline service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerConfigLoader.java:56]: is it ok to use common passwords such as '123456'?

File: service/component-samples/owner/src/main/java/org/fido/iot/sample/OwnerServerApp.java [hunk 4]
```diff
@@ -39,15 +39,23 @@ public static void main(String[] args) {
...
     ctx.addParameter(OwnerAppConstants.DB_PWD,
         OwnerConfigLoader.loadConfig(OwnerAppConstants.DB_PWD));
+
+    ctx.addParameter("db.url",
+        OwnerConfigLoader.loadConfig(OwnerAppConstants.DB_URL));
+    ctx.addParameter("db.user",
+        OwnerConfigLoader.loadConfig(OwnerAppConstants.DB_USER));
+    ctx.addParameter("db.password",
+        OwnerConfigLoader.loadConfig(OwnerAppConstants.DB_PWD));
+
```
Comment (Darshini-Parikh93) [inline service/component-samples/owner/src/main/java/org/fido/iot/sample/OwnerServerApp.java:58]: Could you please help me understand why `db.url`, `db.user` and `db.password` properties were added when `OwnerAppConstants.DB_URL`, `OwnerAppConstants.DB_USER` and `OwnerAppConstants.DB_PWD` are already present?

File: service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerCustomerServlet.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      }
+    }
+
+    DataSource ds = (DataSource) getServletContext().getAttribute("datasource");
+
+    String keySet = new String(req.getInputStream().readAllBytes(), StandardCharsets.US_ASCII);
+
+    PemLoader.loadPublicKeys(keySet);
+    new ResellerDbManager().defineKeySet(ds, keySet, name, Integer.parseInt(id));
+  }
```
Comment (Darshini-Parikh93) [inline service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerCustomerServlet.java:46]: Shouldn't there be a doDelete method implementation here that will call the `deleteKeySet` function of  `ResellerDbManager` class?

**Comment by trbehera** [comment]
> Will there be a separate PR to enable docker deployment for reseller?

Yes, we can take care of that in a separate PR.

File: service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerConfigLoader.java (hunk 3 above)
Comment (trbehera) [reply service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerConfigLoader.java:56]: Raised issue #50 to track this fix.

File: service/http-api-samples/http-api-rv-sample/pom.xml (hunk 2 above)
Comment (trbehera) [reply service/http-api-samples/http-api-rv-sample/pom.xml:44]: Considering these are api samples, should the artifact id be 'api-rv-sample'?

**Review by johneasterday — APPROVED** [review]
I assume that testing covered the following cases:
- deleting a customer already assigned to device(s)
- updating the customer for a device
- customer with multiple key types (i.e. EC384, RSA2048).  All keys are concatenated together in the KEYS column, correct? If so then a length of 4096 will be an issue.

**Pull request merged**